common/util/src/retry.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{thread::sleep, time::Duration};
+
+/// Given an operation retries it successfully sleeping everytime it fails
+/// If the operation succeeds before the iterator runs out, it returns success
+pub fn retry<I, O, T, E>(iterable: I, mut operation: O) -> Result<T, E>
+where
+    I: IntoIterator<Item = Duration>,
+    O: FnMut() -> Result<T, E>,
+{
+    let mut iterator = iterable.into_iter();
+    loop {
+        match operation() {
+            Ok(value) => return Ok(value),
+            Err(err) => {
+                if let Some(delay) = iterator.next() {
+                    sleep(delay);
```

Comment:
nit: I think we usually don't import functions, e.g. use `thread::sleep` instead of `sleep`
@bors-libra delegate+

Reply:
Done